linkerd/app/core/src/serve.rs

```diff
@@ -1,16 +1,18 @@
 use super::accept_error::AcceptError;
-use futures::{try_ready, Future, Poll};
+use futures::{future, try_ready, Future, Poll};
 use linkerd2_drain as drain;
 use linkerd2_error::Error;
 use linkerd2_proxy_core::listen::{Accept, Listen, Serve};
-use tracing::Span;
+use tracing::{debug, Span};
 use tracing_futures::{Instrument, Instrumented};
 
+pub type Task = Box<dyn Future<Item = (), Error = Error> + Send + 'static>;
```

Comment:
Does the boxed future incur any compile-time overhead over spawning in this module?

Reply:
doesn't appear to, but i'm not microoptimizing for compilation